Content.Shared/Power/Generator/SharedGeneratorSystem.cs

```diff
@@ -0,0 +1,18 @@
+﻿namespace Content.Shared.Power.Generator;
+
+/// <summary>
+/// This handles small, portable generators that run off a material fuel.
+/// </summary>
+public abstract class SharedGeneratorSystem : EntitySystem
+{
+    /// <summary>
+    /// Calculates the expected fuel efficiency based on the optimal and target power levels.
+    /// </summary>
+    /// <param name="targetPower">Target power level</param>
+    /// <param name="optimalPower">Optimal power level</param>
+    /// <returns>Expected fuel efficiency as a percentage</returns>
+    public static float CalcFuelEfficiency(float targetPower, float optimalPower)
+    {
+        return MathF.Pow(optimalPower / targetPower, 1.3f);
```

Comment:
Buh? Should this be a constant? Why 1.3?

Reply:
It's a magic number I found during balancing.